src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1120,33 +1289,104 @@ CHIP_ERROR CASESession::SetEffectiveTime(void)
     return ASN1ToChipEpochTime(effectiveTime, mValidContext.mEffectiveTime);
 }
 
-CHIP_ERROR CASESession::HandleErrorMsg(const System::PacketBufferHandle & msg)
+void CASESession::OnSuccessStatusReport()
 {
-    ReturnErrorCodeIf(msg->Start() == nullptr || msg->DataLength() != sizeof(SigmaErrorMsg), CHIP_ERROR_MESSAGE_INCOMPLETE);
+    ChipLogProgress(SecureChannel, "Success status report received. Session was established");
+    mPairingComplete = true;
 
-    static_assert(sizeof(SigmaErrorMsg) == sizeof(uint8_t),
-                  "Assuming size of SigmaErrorMsg message is 1 octet, so that endian-ness conversion is not needed");
+    // Forget our exchange, as no additional messages are expected from the peer
+    mExchangeCtxt = nullptr;
 
-    SigmaErrorMsg * pMsg = reinterpret_cast<SigmaErrorMsg *>(msg->Start());
-    ChipLogError(SecureChannel, "Received error (%d) during CASE pairing process", pMsg->error);
+    // Call delegate to indicate pairing completion
+    mDelegate->OnSessionEstablished();
 
+    mState = kInitialized;
+}
+
+CHIP_ERROR CASESession::OnFailureStatusReport(Protocols::SecureChannel::GeneralStatusCode generalCode, uint16_t protocolCode)
+{
     CHIP_ERROR err = CHIP_NO_ERROR;
-    switch (pMsg->error)
+    switch (protocolCode)
     {
-    case SigmaErrorType::kUnsupportedVersion:
-        err = CHIP_ERROR_UNSUPPORTED_CASE_CONFIGURATION;
+    case kProtocolCodeInvalidParam:
+        err = CHIP_ERROR_INVALID_CASE_PARAMETER;
         break;
 
-    case SigmaErrorType::kInvalidSignature:
-    case SigmaErrorType::kInvalidResumptionTag:
-    case SigmaErrorType::kUnexpected:
-        err = CHIP_ERROR_INVALID_CASE_PARAMETER;
+    case kProtocolCodeNoSharedRoot:
+        err = CHIP_ERROR_CERT_NOT_TRUSTED;
```

Comment:
I think we would benefit from a NO_SHARED_TRUSTED_ROOT which is different than CERT_NO_TRUSTED?

Reply:
I don't think we have an error defined with this value yet. Did you mean that we should add a new error and use that?

Reply:
> Did you mean that we should add a new error and use that?

Yes